tests/test_workflow_api.py

```diff
@@ -247,10 +257,16 @@ async def test_delete_workflow_task(
         assert res.status_code == 201
         wf_id = res.json()["id"]
 
-        workflow = await db.get(Workflow, wf_id)
-        t0 = await task_factory()
-        t1 = await task_factory()
-        t2 = await task_factory()
+        workflow = await client.get(f"api/v1/workflow/{wf_id}")
+        t0 = add_task(client, 0)
+        t1 = add_task(client, 1)
+        t2 = add_task(client, 2)
+
+        payload = {"task_id": t1["id"]}  # FIXME
+        res = await client.post(
+            f"api/v1/workflow/{wf_id}/add-task/",
+            json=payload,
+        )
 
         await workflow.insert_task(t0.id, db=db)
```

Comment:
Remove insert_task calls, and replace with three calls to add-task endpoint (as the one above)

Reply:
Remove refresh

Reply:
Probably
```python
        assert (
            len((await db.execute(select(WorkflowTask))).scalars().all()) == 3
            len((await db.execute(select(WorkflowTask))).scalars().all()) == 3
        )
```
will fail. If so, we can safely remove it